sound/soc/sof/loader.c

```diff
@@ -129,6 +130,82 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 }
 EXPORT_SYMBOL(snd_sof_fw_parse_ext_data);
 
+static size_t snd_sof_ext_man_size(const struct firmware *fw)
+{
+	const struct ext_man_header *head = (void *)fw->data;
+
+	if (fw->size < sizeof(*head))
+		return 0;
```

Comment:
add comment here. do you return zero to deal with older firmware?

Reply:
Here I deal with super small FW, I add a comment:

```
/*
 * assert fw size is big enough to contain extended manifest header,
 * it prevents from reading unallocated memory from `head` in folowing
* step.
*/
```